File: BoardService/src/main/java/hana/simple/boardservice/global/exception/ApplicationException.java
package hana.simple.boardservice.global.exception;

import hana.simple.boardservice.global.exception.constant.ErrorCode;
import lombok.Getter;

public class ApplicationException extends RuntimeException{
    @Getter
    private final ErrorCode errorCode;
    private final String message;

    public ApplicationException(ErrorCode errorCode, String message) {
        this.errorCode = errorCode;
        this.message = message;
    }

    @Override
    public String getMessage() {
        return message == null? errorCode.getMessage() : message;
    }
}

File: UserService/src/main/kotlin/hana/simple/userservice/global/exception/ApplicationException.kt
package hana.simple.userservice.global.exception

import hana.simple.userservice.global.exception.constant.ErrorCode
import org.springframework.http.HttpStatus

class ApplicationException(
    private val errorCode: ErrorCode,
    override val message: String? = null
) : RuntimeException() {


    val getErrorCode: ErrorCode
        get() = errorCode


    val getMessage: String
        get() = message ?: errorCode.message
}
